File: backend/src/main/java/com/godeltech/mastery/backend/repository/GuaranteeRepository.java
package com.godeltech.mastery.backend.repository;

import com.godeltech.mastery.backend.domain.entity.Car;
import com.godeltech.mastery.backend.domain.entity.Guarantee;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface GuaranteeRepository extends JpaRepository<Guarantee, Long> {

  Optional<Guarantee> findByCar(Car car);
}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/repository/GuaranteeRepository.kt
package com.godeltech.mastery.backend.repository

import com.godeltech.mastery.backend.domain.entity.Car
import com.godeltech.mastery.backend.domain.entity.Guarantee
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface GuaranteeRepository : JpaRepository<Guarantee, Long> {

    fun findByCar(car: Car): Optional<Guarantee>
}